Code/Framework/AzToolsFramework/Tests/AssetFileInfoListComparison.cpp

```diff
@@ -198,6 +213,21 @@ namespace UnitTest
 
         }
 
+        void SetAssetFolder(const AZStd::string& folder)
+        {
+            const AZStd::string cacheProjectRootFolder = folder;
+            const AZStd::string cacheRootFolder = cacheProjectRootFolder + AZ_CORRECT_FILESYSTEM_SEPARATOR + AzFramework::PlatformHelper::GetPlatformName(AzFramework::PlatformId::PC);
+
+            // Set the asset folder alias to the temporary directory
+            AZ::IO::FileIOBase::GetInstance()->SetAlias("@assets@", cacheRootFolder.c_str());
+
+            // Set the asset folder to the settings registry
+            auto settingsRegistry = AZ::SettingsRegistry::Get();
+            ASSERT_TRUE(settingsRegistry != nullptr);
+            settingsRegistry->Set(AZ::SettingsRegistryMergeUtils::FilePathKey_CacheProjectRootFolder, cacheProjectRootFolder);
+            settingsRegistry->Set(AZ::SettingsRegistryMergeUtils::FilePathKey_CacheRootFolder, cacheRootFolder);
+        }
```

Comment:
There is a simpler way to do this.
That involves modifying the ToolsTestApplication constructor to set the `/Amazon/AzCore/Bootstrap/project_cache_path` key into the m_SettingsRegistry which is used to override project Cache folder

```
using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString
auto projectCacheRootOverrideKey = FixedValueString::format("%s/project_cache_path", AZ::SettingsRegistryMergeUtils::BootstrapSettingsRootKey);
m_settingsRegistry->Set(projectCacheRootOverrideKey, m_tempDir.GetDirectory());
```

Reply:
This will inject the project_cache_path key into the Settings Registry which will be used to see the `FilePathKey_CacheProjectRootFolder` value which then leads to the FileIOBase setting it's `@assets@` alias to that cache path value

Reply:
Discussed this with @lumberyard-employee-dm directly. Since I'm going to be on vacation soon he is taking ownership of this PR to add this improvement and see this PR through. Thanks @lumberyard-employee-dm .

Reply:
This has been implemented, resolving.